app/services/concerns/xccdf/rules.rb

```diff
@@ -43,6 +43,11 @@ def rule_for(ref_id:)
         @cached_rules[ref_id]
       end
 
+      def value_ids_for(ref_ids:)
+        @cached_values ||= @value_definitions.index_by(&:ref_id)
+        ref_ids.map { |ref_id| @cached_values[ref_id].id }
+      end
```

Comment:
Mixing variable names and memoized instance variables. Please don't do that and follow the existing patterns instead.

Reply:
Also anything cached related to values should go where value definitions are declared. Again...follow existing patterns.

Reply:
I don't understand this comment: 
`Mixing variable names and memoized instance variables. Please don't do that and follow the existing patterns instead.`

Is the issue with this line? 
`ref_ids.map { |ref_id| @cached_values[ref_id].id }`

Reply:
No, the issue is with the whole method, you are mixing existing patterns together. The `@cached_xy` is being used in the `xys_for` method and it is intended to be called multiple times. Here you are mixing it together with the `xy_ids_for` method that is being used with the `@xy` instance variable elsewhere. Also the method name is `value_ids_for` while you're working with `value_definitions`, these should be kept consistent across this codebase.

My suggestion is to extract the `value_definition_id_for` (with a cached instance variable) into a separate method in the `ValueDefinitions` module and use it here + rename this method as well to keep all consistent.